objectRecognitionSource/src/main/java/io/vantiq/extsrc/objectRecognition/ObjectRecognitionCore.java

```diff
@@ -355,28 +356,40 @@ public void sendDataFromImage(ImageRetrieverResults imageResults, ExtensionServi
            return;
        }
        
+       Map<String, Object> processingParams = new LinkedHashMap<>();
+       processingParams.put("timestamp", imageResults.getTimestamp());
+       
        // Send the results of the neural net, or send a Query error if an exception occurs
+       NeuralNetInterface localNeuralNet = null;
        try {
-           synchronized (this) {
+           synchronized (SYNCH_KEY) {
                if (neuralNet == null) { // Should only happen when close() runs just before sendDataFromImage()
                    if (client != null) {
                        client.sendQueryError(replyAddress, this.getClass().getPackage().getName() + ".closed",
                                "The source closed mid message", null);
                    }
                    return;
```

Comment:
would be better to move the client.sendQuery() outside the sync block.  No need to block there.  At some level, it doesn't matter because the source already closed, but better code hygiene to have that outside the synchronized block.